core/src/main/java/hudson/model/UsageStatistics.java

```diff
@@ -123,58 +127,42 @@ private RSAPublicKey getKey() {
      */
     public String getStatData() throws IOException {
         Jenkins j = Jenkins.getInstance();
+        JSONObject data = new JSONObject();
 
-        JSONObject o = new JSONObject();
-        o.put("stat",1);
-        o.put("install", j.getLegacyInstanceId());
-        o.put("servletContainer", j.servletContext.getServerInfo());
-        o.put("version", Jenkins.VERSION);
+        ExtensionList<UsageStatisticsContributor> contributors = UsageStatisticsContributor.all();
 
         List<JSONObject> nodes = new ArrayList<JSONObject>();
-        for( Computer c : j.getComputers() ) {
-            JSONObject  n = new JSONObject();
-            if(c.getNode()==j) {
-                n.put("master",true);
-                n.put("jvm-vendor", System.getProperty("java.vm.vendor"));
-                n.put("jvm-name", System.getProperty("java.vm.name"));
-                n.put("jvm-version", System.getProperty("java.version"));
+        for (Computer computer : j.getComputers()) {
+            JSONObject nodeData = new JSONObject();
+            for (UsageStatisticsContributor contributor : contributors) {
+                try { // robustness: a sole contributor shouldn't be able to interrupt the whole stats retrieval process
+                    contributor.contributeNodeData(computer, nodeData);
+                } catch (Exception e) {
+                    LOGGER.log(Level.WARNING, null, e);
+                }
             }
-            n.put("executors",c.getNumExecutors());
-            DescriptorImpl descriptor = j.getDescriptorByType(DescriptorImpl.class);
-            n.put("os", descriptor.get(c));
-            nodes.add(n);
-        }
-        o.put("nodes",nodes);
-
-        List<JSONObject> plugins = new ArrayList<JSONObject>();
-        for( PluginWrapper pw : j.getPluginManager().getPlugins() ) {
-            if(!pw.isActive())  continue;   // treat disabled plugins as if they are uninstalled
-            JSONObject p = new JSONObject();
-            p.put("name",pw.getShortName());
-            p.put("version",pw.getVersion());
-            plugins.add(p);
+            nodes.add(nodeData);
         }
-        o.put("plugins",plugins);
-
-        JSONObject jobs = new JSONObject();
-        List<TopLevelItem> items = j.getAllItems(TopLevelItem.class);
-        for (TopLevelItemDescriptor d : Items.all()) {
-            int cnt=0;
-            for (TopLevelItem item : items) {
-                if(item.getDescriptor()==d)
-                    cnt++;
+        data.put("nodes", nodes);
+
+        for (UsageStatisticsContributor contributor : contributors) {
+            Class<? extends UsageStatisticsContributor> contributorClass = contributor.getClass();
+            LOGGER.finest(String.format("Calling '%s' usage stats contributor",  contributorClass));
+
+            try { // robustness: a sole contributor shouldn't be able to interrupt the whole stats retrieval process
```

Comment:
Catching `Throwable` generally seems like a giant code smell to me. 
Not sure it would be justified here :-/, it would catch `OutOfMemoryError`s and many important issues. So yeah, this is not robust in the sense it's currently gonna catch everything, but everything that seems reasonable to catch.
So, I'm not really sure we should do this.

Looking at the code, I do see indeed many instances of that pattern, so maybe it's just me not used to it, and it's just normal for Jenkins core code as it's a container that should aggressively protect instances from others. I'll think about it.

Any opinion @jenkinsci/code-reviewers ?